### ISSUE TITLE
Issues with cy.screenshot (locally and in cicd)

### ISSUE THREAD

**Issue by marc-niclas** [cypress-io/cypress#30419]
### Current behavior

I have two different issues with `cy.screenshot`.

### in ci/cd

I'm using `cypress-io/github-action@v6` to run my cypress tests in ci/cd. I notice that when uploading screenshots from the screenshots folder (using upload-artifact) resulting from calling `cy.screenshot` are cut off. While my viewport is 1600x900, the screenshots come out with a width less than 1300, meaning I'm missing the right side of the screenshot. Locally however, my screenshots come out fine. Any advice here? This is disrupting visual regression testing.

### locally 

calling `cy.screenshot` seems to cause some sort of distortion to the test runner. See screenshot of my Electron below. this persists for about 1-2s before going back to the normal test runner. This isn't a showstopper, but definitely annoying.

![image](https://github.com/user-attachments/assets/dd7fd532-4c47-4145-a505-848a4240eb3f)

### Test code to reproduce

unfortunately, since this is in ci/cd, difficult to reproduce.

### Cypress Version

13.6.4

### Node version

20.12.0

### Operating System

macOS 15.0.1

**Comment by jennifer-shehane** [comment]
Could you provide an example application/setup we could run to reproduce the issue?

**Comment by marc-niclas** [comment]
Thanks for your response @jennifer-shehane. I worked with cypress support to get this resolved. Essentially, my viewport width (1600) exceeded the width of the browser (1280) in ci/cd. By reducing my viewport size, I was able to address this issue.

However, as I told cypress support over email, this still feel likes a bug. `cy.screenshot` size should be limited by either the browser or viewport width, whichever is less.
